File: clients/python/tests/test_smoke.py
# Copyright 2026 Nikolay Samokhvalov. Apache-2.0 license.

"""End-to-end smoke: connect, subscribe, send, tick, receive, ack."""

import pgque


def test_smoke_send_receive_ack(dsn, queue_name, consumer_name):
    with pgque.connect(dsn) as client:
        client.conn.execute("select pgque.create_queue(%s)", (queue_name,))
        client.conn.execute("select pgque.subscribe(%s, %s)",
                             (queue_name, consumer_name))
        client.conn.commit()

        try:
            client.send(queue_name, {"hello": "world"}, type="smoke.test")
            client.conn.commit()

            client.conn.execute("select pgque.force_tick(%s)", (queue_name,))
            client.conn.execute("select pgque.ticker()")
            client.conn.commit()

            msgs = client.receive(queue_name, consumer_name, max_messages=10)
            assert len(msgs) == 1
            assert msgs[0].type == "smoke.test"
            assert msgs[0].payload == {"hello": "world"} or \
                   msgs[0].payload == '{"hello": "world"}'

            client.ack(msgs[0].batch_id)
            client.conn.commit()
        finally:
            client.conn.rollback()
            client.conn.execute(
                "select pgque.unregister_consumer(%s, %s)",
                (queue_name, consumer_name))
            client.conn.execute("select pgque.drop_queue(%s, true)",
                                 (queue_name,))
            client.conn.commit()

File: clients/ruby/test/test_smoke.rb
# Copyright 2026 Nikolay Samokhvalov. Apache-2.0 license.

require_relative "test_helper"

class TestSmoke < Minitest::Test
  include PgqueTest::Helpers

  def test_smoke_send_receive_ack
    queue = unique_queue_name
    consumer_n = unique_consumer_name
    Pgque.connect(dsn) do |client|
      client.conn.exec_params("select pgque.create_queue($1)", [queue])
      client.conn.exec_params("select pgque.subscribe($1, $2)", [queue, consumer_n])

      begin
        client.send(queue, { "hello" => "world" }, type: "smoke.test")
        client.conn.exec_params("select pgque.force_next_tick($1)", [queue])
        client.conn.exec_params("select pgque.ticker($1)", [queue])

        msgs = client.receive(queue, consumer_n, 10)
        assert_equal 1, msgs.size
        assert_equal "smoke.test", msgs[0].type
        assert_equal({ "hello" => "world" }, msgs[0].payload)

        client.ack(msgs[0].batch_id)
      ensure
        client.conn.exec_params("select pgque.unregister_consumer($1, $2)",
                                [queue, consumer_n]) rescue nil
        client.conn.exec_params("select pgque.drop_queue($1, true)", [queue]) rescue nil
      end
    end
  end
end
